src/main/java/com/minecolonies/core/entity/ai/minimal/EntityAIEatTask.java

```diff
@@ -292,7 +291,7 @@ private EatingState goToEatingPlace()
             }
         }
 
-        if (EntityNavigationUtils.walkToPos(citizen, eatPos, 2, true))
+        if (citizen.getVehicle() != null || EntityNavigationUtils.walkToPos(citizen, eatPos, 2, true))
```

Comment:
this shouldnt be needed, if it is then another logic is failing, an it creates a new sitting entity for the citizen that is sitting already

Reply:
The citizen end up in a sit/stand loop because walkToPos resets the sitting position